lewis/examples/example_motor/__init__.py

```diff
@@ -97,11 +97,14 @@ class ExampleMotorStreamInterface(StreamAdapter):
     The motor starts moving immediately when a new target position is set. Once it's moving,
     it has to be stopped to receive a new target, otherwise an error is generated.
     """
+
+    pattern_type = fmt
+
     commands = {
-        Cmd('get_status', r'^S\?$'),
-        Cmd('get_position', r'^P\?$'),
-        Cmd('get_target', r'^T\?$'),
-        Cmd('set_target', r'^T=([-+]?[0-9]*\.?[0-9]+)$', argument_mappings=(float,)),
+        Cmd('get_status', regex(r'^S\?$')),
+        Cmd('get_position', r'^P?$'),
+        Cmd('get_target', r'^T?$'),
```

Comment:
I checked the regex, it's `^T\\?$` (or `r'^T\?$'`). 

This is really unexpected, isn't it? 

In scanf format, `$` has no special meaning and should match literal `$`. `^` only has a special meaning as a not in `[^...]`, outside of that I think it should also match a literal `^`. They certainly shouldn't match beginning and end of string, as they do here.

I understand it's work in progress, just thought this would be important to point out. It's a limitation that may well affect how we should design this feature as a whole.